js/output/pjs/babyhint.js

```diff
@@ -641,10 +687,23 @@ var BabyHint = {
             }
 
             // count the parameters passed
-            var numParams;
-            var numCommas = params.match(/,/g);
-            if (numCommas) {
-                numParams = numCommas.length + 1;
+            var numParams, rawCount = this.countCommas(params.substring(1, params.length - 1));
+
+            // push in an error if an error occurs while parsing the param list
+            if (rawCount.error) {
+                errors.push({
+                    row: lineNumber,
+                    column: index + 2 + rawCount.error.index,
+                    text: rawCount.error.message,
+                    breaksCode: false,
+                    source: "paramschecker",
+                    context: {}
```

Comment:
The tests don't seem to cover the two possible errors from the paramschecker, from what I can tell, is that right? Can you add tests that cover those errors?

Reply:
Ok, I added some more tests.